gcpvault_test.go

```diff
@@ -206,6 +262,11 @@ func TestGetSecrets(t *testing.T) {
 			}
 
 			iamSvr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if test.wantIAMRetry {
+					test.wantIAMRetry = false
```

Comment:
Why is setting this to false needed? I'm not sure I understand how the retry logic works, can you explain?  Does this mean only one request will fail?

Reply:
`wantIAMRetry = true` makes the first request fail and then it will retry.

Reply:
Ah ok, thx.